pkg/build/controller/build/build_controller.go

```diff
@@ -372,10 +372,15 @@ func (bc *BuildController) handleBuild(build *buildapi.Build) error {
 // However if the build is either complete or failed and its completion timestamp
 // has not been set, then it returns false so that the build's completion timestamp
 // gets updated.
-func shouldIgnore(build *buildapi.Build) bool {
-	// If pipeline build, do nothing.
+func (bc *BuildController) shouldIgnore(build *buildapi.Build) bool {
+	// If pipeline build, only handle pruning.
 	// These builds are processed/updated/etc by the jenkins sync plugin
 	if build.Spec.Strategy.JenkinsPipelineStrategy != nil {
+		if buildutil.IsBuildComplete(build) {
+			if err := common.HandleBuildPruning(buildutil.ConfigNameForBuild(build), build.Namespace, bc.buildLister, bc.buildConfigGetter, bc.buildDeleter); err != nil {
+				utilruntime.HandleError(fmt.Errorf("failed to prune builds for %s/%s: %v", build.Namespace, build.Name, err))
+			}
+		}
```

Comment:
this isn't the right place for this... shouldIgnore should not have side-effects.